### PR TITLE
Expose `rz_core_reg_assign_sync` and `rz_core_reg_filter_items_sync` API

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

expose `rz_core_reg_assign_sync()` and `rz_core_reg_filter_items_sync()`

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/core/cmd/cmd_regs.c [hunk 1]
```diff
@@ -124,6 +124,32 @@ static void print_reg_not_found(const char *arg) {
 	RZ_LOG_ERROR("No such register or register type: \"%s\"\n", rz_str_get(arg));
 }
 
+/**
+ * Register assignment like reg=0x42
+ *
+ * \param core The RzCore instance
+ * \param reg The RzReg instance
+ * \param sync_cb Callback for synchronizing register state in command
+ * \param name Reg name
```
Comment (XVilka) [inline librz/core/cmd/cmd_regs.c]: `Register name`

**Review by XVilka — APPROVED** [review]

File: librz/core/cmd/cmd_regs.c [hunk 2]
```diff
@@ -124,6 +124,32 @@ static void print_reg_not_found(const char *arg) {
...
+ * Register assignment like reg=0x42
+ *
+ * \param core The RzCore instance
+ * \param reg The RzReg instance
+ * \param sync_cb Callback for synchronizing register state in command
+ * \param name Register name
+ * \param val Value
+ * \return success
+ */
+RZ_API bool rz_core_reg_assign_sync(RZ_NONNULL RzCore *core, RZ_NONNULL RzReg *reg, RzCmdRegSync sync_cb, RZ_NONNULL const char *name, ut64 val) {
```
Comment (wargio) [inline librz/core/cmd/cmd_regs.c:137]: is it the callback even used?

Comment (XVilka) [reply librz/core/cmd/cmd_regs.c:137]: @imbillow

Comment (b1llow) [reply librz/core/cmd/cmd_regs.c:137]: > is it the callback even used?

it used when rz_cmd

```
static bool cmd_regs_sync(RzCore *core, RzRegisterType type, bool write) {
	return rz_debug_reg_sync(core->dbg, type, write);
}
```

Comment (wargio) [reply librz/core/cmd/cmd_regs.c:137]: i do not see it called in this function nor passed to any function.

Comment (b1llow) [reply librz/core/cmd/cmd_regs.c:137]: I mean this chain, in the `rz_debug_regs_handler`, it pass the `rz_debug_reg_sync` as the `sync_cb`

https://github.com/rizinorg/rizin/blob/7fa2c132de30f7cd1e73c74a3eab161dad8e33fc/librz/core/cmd/cmd_regs.c#L282-L289
https://github.com/rizinorg/rizin/blob/7fa2c132de30f7cd1e73c74a3eab161dad8e33fc/librz/core/cmd/cmd_regs_meta.inc#L18-L19
https://github.com/rizinorg/rizin/blob/7fa2c132de30f7cd1e73c74a3eab161dad8e33fc/librz/core/cmd/cmd_debug.c#L3909-L3913

Comment (thestr4ng3r) [reply librz/core/cmd/cmd_regs.c:137]: @wargio see the definitions of `SYNC_READ` and `SYNC_WRITE`

**Review by wargio — APPROVED** [review]

**Pull request merged**